### ISSUE TITLE
Videos have incorrect time on timeline

### ISSUE THREAD

**Issue by LeonardFiedrowicz** [pulsejet/memories#815]
**Describe the bug**
Video files show up in the wrong order in the timeline, only during summertime. The time at the info tab doesn't match up with the real time it was taken at and doesn't match up with ordering in the timeline either. Depening on the camera it's either sorted correctly but displays the wrong time in the info tab or displays the wrong time  at -1h but gets sorted as +1h. Files from before March during wintertime get ordered correctly.

**To Reproduce**
Check file order of a video and a corresponding photo from the same time. The photo has the correct timestamp, the video doesn't and gets ordered incorrectly. (sample files sent via Discord)

**Platform:**

- OS: Windows
- Browser: Firefox, Edge
- Memories Version: 5.4.1
- Nextcloud Version: 27.0.2.  AIO Version 27.1.1 beta
- PHP Version: 8.1.23

- Any errors in the Nextcloud server logs?
Probably not related but I found this:
[webdav] Fehler: Sabre\DAV\Exception: Unknown error while seeking content at <<closure>>

0. /var/www/html/apps/dav/lib/BulkUpload/MultipartRequestParser.php line 129
   OCA\DAV\BulkUpload\MultipartRequestParser->isAt("--boundary_.oOo ... n")
1. /var/www/html/apps/dav/lib/BulkUpload/BulkUploadPlugin.php line 71
   OCA\DAV\BulkUpload\MultipartRequestParser->isAtLastBoundary()
2. /var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
   OCA\DAV\BulkUpload\BulkUploadPlugin->httpPost(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
3. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 472
   Sabre\DAV\Server->emit("method:POST", [["Sabre\\HTTP\\ ... ]])
4. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 253
   Sabre\DAV\Server->invokeMethod(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
5. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 321
   Sabre\DAV\Server->start()
6. /var/www/html/apps/dav/lib/Server.php line 364
   Sabre\DAV\Server->exec()
7. /var/www/html/apps/dav/appinfo/v2/remote.php line 35
   OCA\DAV\Server->exec()
8. /var/www/html/remote.php line 172
   require_once("/var/www/html/a ... p")

POST /remote.php/dav/bulk
from 83.135.74.59 by EFA at 2023-09-11T13:10:00+00:00

**Comment by pulsejet** [comment]
So I took a look at these; the fundamental issue with videos is that they don't carry any timezone information whatsoever, so it is impossible to infer the exact time. Now, Memories works around this in a few ways:
1. It tries to find the timezone anyway, some modern cameras may add this.
1. If no extra information is found, it tries to use the server's timezone.
1. If available it uses the GPS location to get the timezone for the video

Basically, the (only) way around this issue is to manually tag the videos with the GPS location whenever absent. I tried this for a couple of videos and it seemed to correct the timezone and sorting issues.

Also the reason the issue happens during some parts of the year and not others is likely related to daylight savings.

**Comment by veenstrajelmer** [comment]
@pulsejet I am not sure if your answer above is still up to date. I am encountering the same issue. If I take a photo, then a video at least one minute later, and then a photo at least one minute later. I expect them to show up in this order in memories. However, they are ordered video-photo-photo. My timezone is UTC+2 (CEST). It seems as if the video metadata is recomputed to UTC, making it end up two hours before the photos instead of at approximately the same time. What would make more sense to me is that the local time would be used (without correction), just like you mentioned recently in https://github.com/pulsejet/memories/issues/1495#issuecomment-3148552991.
